domain/secret/state/state.go

```diff
@@ -1409,36 +1429,35 @@ SELECT sm.secret_id AS &secretInfo.secret_id,
        so.owner_id,
        so.label) AS (&secretOwner.*)
 FROM   secret_metadata sm
-       JOIN secret_revision sr ON sm.secret_id = sr.secret_id
-       LEFT JOIN secret_revision_expire sre ON sre.revision_uuid = sr.uuid
-       LEFT JOIN secret_rotate_policy rp ON rp.id = sm.rotate_policy_id
-       LEFT JOIN secret_rotation sro ON sro.secret_id = sm.secret_id
-       LEFT JOIN (
-          SELECT $ownerKind.model_owner_kind AS owner_kind, (SELECT uuid FROM model) AS owner_id, label, secret_id
-          FROM   secret_model_owner so
-          UNION
-          SELECT $ownerKind.application_owner_kind AS owner_kind, application.name AS owner_id, label, secret_id
-          FROM   secret_application_owner so
-          JOIN   application
-          WHERE  application.uuid = so.application_uuid
-          UNION
-          SELECT $ownerKind.unit_owner_kind AS owner_kind, unit.name AS owner_id, label, secret_id
-          FROM   secret_unit_owner so
-          JOIN   unit
-          WHERE  unit.uuid = so.unit_uuid
-       ) so ON so.secret_id = sm.secret_id`
+JOIN secret_revision AS sr ON sm.secret_id = sr.secret_id
+LEFT JOIN secret_revision_expire AS sre ON sre.revision_uuid = sr.uuid
+LEFT JOIN secret_rotate_policy AS rp ON rp.id = sm.rotate_policy_id
+LEFT JOIN secret_rotation AS sro ON sro.secret_id = sm.secret_id
+LEFT JOIN secret_owner AS so ON so.secret_id = sm.secret_id
+`
 
 	queryTypes := []any{
 		secretInfo{},
 		secretOwner{},
 		ownerKindParam,
 	}
 	queryParams := []any{ownerKindParam}
+	if uri != nil && label != nil {
+		return nil, errors.New("cannot specify both URI and label")
+	}
 	if uri != nil {
 		queryTypes = append(queryTypes, secretID{})
 		query = query + "\nWHERE sm.secret_id = $secretID.id"
 		queryParams = append(queryParams, secretID{ID: uri.ID})
 	}
+	type secretLabelOrId struct {
+		Value string `db:"value"`
+	}
+	if label != nil {
+		queryTypes = append(queryTypes, secretLabelOrId{})
+		query = query + "\nWHERE so.label = $secretLabelOrId.value OR sm.secret_id = $secretLabelOrId.value"
```

Comment:
This isn't right - a label is not a secret id. The label and id (secret uri) attributes are distinct and different from each other. The label filter term is just used to march a secret label.